### ISSUE TITLE
`FlaxDDPMScheduler` produces black images

### ISSUE THREAD

**Issue by pcuenca** [huggingface/diffusers#2179]
### Describe the bug

`FlaxDDPMScheduler` produces black images, even after fixing #2176.

I checked in `bfloat16`, `bfloat32` and at revision f106ab40. It failed in all cases. 

### Reproduction

```Python
import jax
import numpy as np
from flax.jax_utils import replicate
from flax.training.common_utils import shard

from diffusers import FlaxStableDiffusionPipeline, FlaxDDPMScheduler

model_id = "runwayml/stable-diffusion-v1-5"
scheduler, scheduler_state = FlaxDDPMScheduler.from_pretrained(
    model_id,
    subfolder="scheduler"
)
pipeline, params = FlaxStableDiffusionPipeline.from_pretrained(
    model_id,
    scheduler=scheduler,
    revision="bf16",
    dtype=jax.numpy.bfloat16,
)
params["scheduler"] = scheduler_state

prompt = "a photo of an astronaut riding a horse on mars"
prng_seed = jax.random.PRNGKey(0)
num_inference_steps = 50

num_samples = jax.device_count()
prompt = num_samples * [prompt]
prompt_ids = pipeline.prepare_inputs(prompt)

params = replicate(params)
prng_seed = jax.random.split(prng_seed, jax.device_count())
prompt_ids = shard(prompt_ids)

images = pipeline(prompt_ids, params, prng_seed, num_inference_steps, jit=True).images
images = pipeline.numpy_to_pil(np.asarray(images.reshape((num_samples,) + images.shape[-3:])))

for i, image in enumerate(images):
    image.save(f"flax_{i}.png")
```

### System Info

Diffusers `0.13.0.dev0`. Also tested on previous revisions as indicated above.

**Comment by pcuenca** [comment]
Initially reported as part of #2155.

**Comment by patrickvonplaten** [comment]
Interesting, are we sure that it's not a bug that exists from the very beginning? If DDIM and DPM works maybe this one is also not super important to fix?

**Comment by kxhit** [comment]
I found a similar bug when I set bf16. I traced the bug and found the scheduler.step() will return .prev_sample full of nan, while the value is correct before returning the scheduler.step(). No idea why.
